umm-spec-lib/src/cmr/umm_spec/location_keywords.clj

```diff
@@ -43,20 +43,15 @@
 (defn- location-values
   "Returns the location keyword values in order so that we can get the last one"
   [location-keyword]
-  (for [k location-keyword-order
-        :let [value (get location-keyword k)]
-        :when value]
-    value))
-
-(defn- leaf-value
-  "Returns the leaf value of the location-keyword object to be put in a SpatialKeywords list"
-  [location-keyword]
-  (last (location-values location-keyword)))
+  (as-> location-keyword keyword
+        (vals keyword)
+        (distinct keyword)
+        (remove nil? keyword)))
 
 (defn location-keywords->spatial-keywords
   "Converts a list of LocationKeyword maps to a list of SpatialKeywords"
   [location-keyword-list]
-  (map #(leaf-value %) location-keyword-list))
+  (flatten (map location-values location-keyword-list)))
```

Comment:
This could cause a major behavior change for groups outside the MMT. I'm trying to remember why we only wanted to match the last value in a location keyword.

Give me a little time to double-check what impact this will have.